lib/v2/screens/sign_up/viewmodels/states/create_username_state.dart

```diff
@@ -1,33 +1,59 @@
 import 'package:equatable/equatable.dart';
+import 'package:seeds/utils/string_extension.dart';
+import 'package:seeds/v2/domain-shared/page_command.dart';
 import 'package:seeds/v2/domain-shared/page_state.dart';
 
+class UsernameGenerated extends PageCommand {}
+
 class CreateUsernameState extends Equatable {
   final PageState pageState;
+  final PageCommand? pageCommand;
   final String? username;
-  final bool isValidUsername;
+  final String? errorMessage;
 
   const CreateUsernameState({
     required this.pageState,
-    required this.isValidUsername,
+    this.pageCommand,
     this.username,
+    this.errorMessage,
   });
 
   @override
-  List<Object?> get props => [pageState, isValidUsername];
+  List<Object?> get props => [pageState, username, errorMessage];
 
   CreateUsernameState copyWith({
     PageState? pageState,
+    PageCommand? pageCommand,
     String? username,
-    bool? isValidUsername,
+    String? errorMessage,
   }) {
     return CreateUsernameState(
       pageState: pageState ?? this.pageState,
+      pageCommand: pageCommand ?? this.pageCommand,
       username: username ?? this.username,
-      isValidUsername: isValidUsername ?? this.isValidUsername,
+      errorMessage: errorMessage,
     );
   }
 
   factory CreateUsernameState.initial() {
-    return const CreateUsernameState(pageState: PageState.initial, isValidUsername: false);
+    return const CreateUsernameState(
+      pageState: PageState.initial,
+      username: null,
+    );
   }
+
+  factory CreateUsernameState.loading(CreateUsernameState currentState) {
+    return currentState.copyWith(pageState: PageState.loading, errorMessage: null);
+  }
+
+  factory CreateUsernameState.error(CreateUsernameState currentState, String errorMessage) {
+    return currentState.copyWith(
+      pageState: PageState.failure,
+      errorMessage: errorMessage,
+    );
+  }
+
+  bool get isUsernameValid => !username.isNullOrEmpty && pageState == PageState.success;
+
+  bool get isNextButtonActive => isUsernameValid;
```

Comment:
this is silly haha. I meant that the logic in the UI should not be there. But you could use `isUsernameValid` to know if the button is available or not. This works too tho.

Reply:
This is not a business logic though, it's just showing or hiding a button. I think cases like this can be getters in BLoC to just keep the UI clean